src/Preg.php

```diff
@@ -36,12 +36,12 @@
     public static function match($pattern, $subject, &$matches = null, $flags = 0, $offset = 0)
     {
         $result = @preg_match(self::addUtf8Modifier($pattern), $subject, $matches, $flags, $offset);
-        if (false !== $result) {
+        if (false !== $result && PREG_NO_ERROR === preg_last_error()) {
```

Comment:
Perhaps this logic can be split into a private method like `isErroneousResult($result)`?

Reply:
I was thinking about it, but `$result` would have to be compared sometimes to `false` and sometimes to `null` - having both in the private method would be confusing for the reader and having another argument in method call was looking less readable either.

Reply:
Makes sense